src/client.ts

```diff
@@ -0,0 +1,217 @@
+const MAX_SEEN_TRACKING = 2048;
+
+export type StartStreamResponse = {
+  token: string;
+  channel_id: string;
+  algorithm: string;
+  expires_in: number;
+};
+
+export type BaseEnvelope = {
+  event_type: 'stream' | 'control';
+  message_id: string;
+};
+
+// To match the rust enums
+export type StreamPhase = 'PHASE_START' | 'PHASE_DELTA' | 'PHASE_END' | 'PHASE_ERROR';
+
+export type StreamEnvelope<T> = BaseEnvelope & {
+  event_type: 'stream';
+  sequence: number;
+  phase: StreamPhase;
+  payload?: T;
+};
```

Comment:
```suggestion
export interface StreamEnvelope<T> extends BaseEnvelope {
  event_type: 'stream';
  sequence: number;
  phase: StreamPhase;
  payload?: T;
};
```

slightly better for type performance

Reply:
this is more of a nit, doesn't really matter

Reply:
gonna punt for right now